File: scripts/test_solutions/309_best_time_buy_sell_cooldown.js
var maxProfit = function(prices) {
    let sold = 0, held = -Infinity, rest = 0;
    for (const p of prices) {
        const prevSold = sold;
        sold = held + p;
        held = Math.max(held, rest - p);
        rest = Math.max(rest, prevSold);
    }
    return Math.max(sold, rest);
};

File: scripts/test_solutions/309_best_time_buy_sell_cooldown.py
def maxProfit(prices):
    sold, held, rest = 0, float('-inf'), 0
    for p in prices:
        prev_sold = sold
        sold = held + p
        held = max(held, rest - p)
        rest = max(rest, prev_sold)
    return max(sold, rest)
